### ISSUE TITLE
Stability of textlint-rule-no-dead-link 

### ISSUE THREAD

**Issue by azu** [textlint/textlint#453]
textlint-rule-no-dead-link is introduced by #442 
This rule includes false-positive in principle.
Because, this rule depended on server status.

We have some options for resolving this issue.

- Make `no-dead-link`'s severity  `"warning"`
  - Simple option
- Enhance textlint-rule-no-dead-link it-self
  - Improve timeout error handling
- Report lint result by comment instead of exit status
  - It can prevent to fail CI. Instead of, a bot comment the lint results.
  - [Danger JS](http://danger.systems/js/ "Danger JS") and [reviewdog](https://github.com/haya14busa/reviewdog "reviewdog") help this option

**Comment by 0x6b** [comment]
> Enhance textlint-rule-no-dead-link it-self

Just an idea. 40x ([client error](https://tools.ietf.org/html/rfc7231#section-6.5)) should be fixed immediately while 50x could be ignored because (some of) it indicates intermittent error on server side. Specify severity by its HTTP status code might reduce false–positive?

**Comment by Leko** [comment]
I agree with @0x6b idea to reduce false–positive.

- 50x: Not error
- 40x: Error
- (301: warning?)

**Comment by azu** [comment]
[textlint-rule-no-dead-link](https://github.com/textlint-rule/textlint-rule-no-dead-link) aim to find dead link.
5xx response is a part of dead link. The reason come from server.

For example, If a server response "Internal Error 500" for a long time, this link may be dead.

If the 5xx is **temporary** error, the rule can ignore it.
But, the rule(client side) can not know the true.

> (301: warning?)

This is a fixable error.
https://github.com/textlint-rule/textlint-rule-no-dead-link#obsolete-link-detection

**Comment by Leko** [comment]
Hmm...
Then, I think it's better: `Report the results with comments instead of the end state`.
If error mute as warning, it prevent false-positive but I think it is dead letter.

It is to take time on each PR, so probably we don't look these warnings.
It will increase the build time meanlessly.

**Comment by melg8** [comment]
Encountered same issue with some runs triggering:
``` bash
  62:16  error  https://github.com/textlint/textlint is dead. (429 too many requests)            no-dead-link
```
and some times - clean run.

**Comment by azu** [comment]
I've written a recipe: [textlint + SARIF output + code scanning](https://github.com/textlint-rule/textlint-rule-no-dead-link#textlint--sarif-output--code-scanning)
